src/main/scala/it/pagopa/interop/apigateway/api/impl/AuthApiServiceImpl.scala

```diff
@@ -0,0 +1,130 @@
+package it.pagopa.interop.apigateway.api.impl
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.{complete, onComplete}
+import akka.http.scaladsl.server.Route
+import cats.implicits._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.apigateway.api.AuthApiService
+import it.pagopa.interop.apigateway.common.ApplicationConfiguration
+import it.pagopa.interop.apigateway.error.GatewayErrors._
+import it.pagopa.interop.apigateway.model.TokenType.Bearer
+import it.pagopa.interop.apigateway.model.{ClientCredentialsResponse, Problem}
+import it.pagopa.interop.apigateway.service.{AuthorizationManagementInvoker, AuthorizationManagementService}
+import it.pagopa.interop.authorizationmanagement.client.model.{
+  Client,
+  ClientComponentState,
+  ClientKind,
+  ClientStatesChain
+}
+import it.pagopa.pdnd.interop.commons.jwt.model.{ClientAssertionChecker, RSA, ValidClientAssertionRequest}
+import it.pagopa.pdnd.interop.commons.jwt.service.{ClientAssertionValidator, PDNDTokenGenerator}
+import it.pagopa.pdnd.interop.commons.jwt.{JWTConfiguration, JWTInternalTokenConfig}
+import it.pagopa.pdnd.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.pdnd.interop.commons.utils.TypeConversions._
+import org.slf4j.LoggerFactory
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class AuthApiServiceImpl(
```

Comment:
I would agree on the `final`, but IMHO it's useless to promote it to case class since we'll never access its fields or compare it to any other instance. This class is a service like one, not a data like one.